src/main/java/elrond/esdt/common/TokenIdentifier.java

```diff
@@ -0,0 +1,41 @@
+package elrond.esdt.common;
+
+import org.apache.commons.codec.DecoderException;
+import org.apache.commons.codec.binary.Hex;
+
+import java.nio.charset.StandardCharsets;
+
+public class TokenIdentifier {
+    private TokenIdentifier() {}
+    private String hexTokenIdentifier;
+
+    public static TokenIdentifier fromString(String tokenIdentifier) {
+        TokenIdentifier ti = new TokenIdentifier();
+        ti.hexTokenIdentifier = Hex.encodeHexString(tokenIdentifier.getBytes(StandardCharsets.UTF_8));
+
+        return ti;
+    }
+
+    public static TokenIdentifier fromHex(String hexTokenIdentifier) {
+        TokenIdentifier ti = new TokenIdentifier();
+        ti.hexTokenIdentifier = hexTokenIdentifier;
+
+        return ti;
+    }
+
+    @Override
+    public String toString() {
+        byte[] bytes = new byte[0];
+        try {
+            bytes = Hex.decodeHex(this.hexTokenIdentifier.toCharArray());
+        } catch (DecoderException e) {
+            e.printStackTrace();
```

Comment:
So we log the stack trace, but swallow the exception? Oh, because it's `toString()` and we cannot attach `throws` to it, right?

Reply:
Anyway, the `hexTokenIdentifier` is always valid (private member etc.)

Reply:
renamed to `toRegularString()` in order to avoid the overloading of the Object class